src/dbnode/persist/fs/migration/migration.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package migration
+
+import (
+	"github.com/m3db/m3/src/dbnode/namespace"
+	"github.com/m3db/m3/src/dbnode/persist"
+	"github.com/m3db/m3/src/dbnode/persist/fs"
+	"github.com/m3db/m3/src/dbnode/storage"
+	xtime "github.com/m3db/m3/src/x/time"
+)
+
+// Migration interface is implemented by tasks that wish to perform a data migration
+// on a fileset. Typically involves updating files in a fileset that were created by
+// a previous version of the database client
+type Task interface {
+
+	// Run is the set of steps to successfully complete a migration
+	Run() error
+}
+
+// ToVersion1_1Task is an object responsible for migrating a fileset to version 1.1
+type ToVersion1_1Task struct {
+	newMergerFn       fs.NewMergerFn
+	infoFileResult    fs.ReadInfoFileResult
+	shard             uint32
+	namespaceMetadata namespace.Metadata
+	persistManager    persist.Manager
+	sOpts             storage.Options
+	fsOpts            fs.Options
+}
+
+// ShouldMigrateToVersion1_1 returns true or false depending on if a fileset should be migrated to 1.1 or not
```

Comment:
nit: `// ShouldMigrateToVersion1_1 indicates whether the fileset should be migrated to 1.1 or not` maybe?

Reply:
Sometimes writing the comments are the hardest part tbh ;)